jerry-ext/arg/arg-transform-functions.c

```diff
@@ -84,14 +87,160 @@ jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< avail
                                (jerry_char_t *) "It can not be converted to a number.");
   }
 
-  double *dest = c_arg_p->dest;
-  *dest = jerry_get_number_value (to_number);
+  *number_p = jerry_get_number_value (to_number);
   jerry_release_value (to_number);
 
   return jerry_create_undefined ();
+} /* jerryx_arg_transform_number_common */
+
+/**
+ * Tranform a JS argument to a double. Type coercion is not allowed.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+jerry_value_t
+jerryx_arg_transform_number_strict (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< available JS args */
+                                    const jerryx_arg_t *c_arg_p) /**< the native arg */
+{
+  return jerryx_arg_transform_number_strict_common (js_arg_iter_p,
+                                                    c_arg_p->dest);
+} /* jerryx_arg_transform_number_strict */
+
+/**
+ * Tranform a JS argument to a double. Type coercion is allowed.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+jerry_value_t
+jerryx_arg_transform_number (jerryx_arg_js_iterator_t *js_arg_iter_p, /**< available JS args */
+                             const jerryx_arg_t *c_arg_p) /**< the native arg */
+{
+  return jerryx_arg_transform_number_common (js_arg_iter_p,
+                                             c_arg_p->dest);
 } /* jerryx_arg_transform_number */
 
 /**
+ * Clamp a double by given min and max value.
+ *
+ * @return clamped double value
+ */
+static double
+jerryx_arg_helper_clamp (double d, /**< the double value */
+                         double min, /**< min value */
+                         double max) /**< max value */
+{
+  const double t = d < min ? min : d;
+  return t > max ? max : t;
+} /* jerryx_arg_helper_clamp */
+
+/**
+ * Round a double.
+ *
+ * @return the rounded double
+ */
+static double
+jerryx_arg_helper_round (double x) /**< the number need to round */
+{
+  return (x >= 0.0) ? floor (x + 0.5) : ceil (x - 0.5);
+} /* jerryx_arg_helper_round */
+
+/**
+ * Helper function to process a double number before converting it
+ * to an integer.
+ *
+ * @return jerry undefined: the transformer passes,
+ *         jerry error: the transformer fails.
+ */
+static jerry_value_t
+jerryx_arg_helper_process_double (double *d, /**< [in, out] the number to be processed */
+                                  double min, /**< the min value for clamping */
+                                  double max, /**< the max value for clamping */
+                                  jerryx_arg_int_option_t option) /**< the converting policies */
+{
+  if (option.clamp == JERRYX_ARG_NO_CLAMP)
+  {
+    if (*d > max || *d < min)
+    {
+      return jerry_create_error (JERRY_ERROR_TYPE,
+                                 (jerry_char_t *) "The number is out of range.");
+    }
+  }
+  else
```

Comment:
Does it make sense to perfect the above code after we impl the assert/unreachable/ndebug such things in jerryx?